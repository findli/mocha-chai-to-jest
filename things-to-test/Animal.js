import React from 'react';

export default class Animal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            clicked: false,
        }
        this.clickEvent = this.pvtClickEvent.bind(this);
    }

    pvtClickEvent(evt) {
        this.setState({
            clicked: true,
        });
    }

    render() {
        const { clicked } = this.state;
        return (
            <div className="section">
                {clicked ? (
                    <p>
                        Woof!
                    </p>
                ) : (
                    <button onClick={this.clickEvent}>
                        Pat me!
                    </button>
                )}
            </div>
        );
    }
}
