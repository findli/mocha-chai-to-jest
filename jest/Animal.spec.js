import React from 'react';
import Animal from '../things-to-test/Animal';
import { shallow } from 'enzyme';

describe('Animal component', () => {
    it('renders correctly with min params', () => {
        const component = shallow(
            <Animal />
        );
        expect(component).toMatchSnapshot();
    });

    it('click event works', () => {
        const component = shallow(
            <Animal />
        );
        expect(component).toMatchSnapshot();

        // we simulate a click
        component.find('button').first().simulate('click');

        // and generate a new snapshot of the component
        expect(component).toMatchSnapshot();
    });
});
