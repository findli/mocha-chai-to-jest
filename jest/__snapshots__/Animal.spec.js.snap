// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal component click event works 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="section"
>
    <button
        onClick={[Function]}
    >
        Pat me!
    </button>
</div>,
  "nodes": Array [
    <div
      className="section"
>
      <button
            onClick={[Function]}
      >
            Pat me!
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Animal />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Animal {
        "_reactInternalInstance": [Circular],
        "clickEvent": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "clicked": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="section"
>
          <button
                    onClick={[Function]}
          >
                    Pat me!
          </button>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="section"
>
          <button
                    onClick={[Function]}
          >
                    Pat me!
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Animal />,
}
`;

exports[`Animal component click event works 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="section"
>
    <p>
        Woof!
    </p>
</div>,
  "nodes": Array [
    <div
      className="section"
>
      <p>
            Woof!
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Animal />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Animal {
        "_reactInternalInstance": [Circular],
        "clickEvent": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "clicked": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="section"
>
          <p>
                    Woof!
          </p>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="section"
>
          <p>
                    Woof!
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Animal />,
}
`;

exports[`Animal component renders correctly with min params 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="section"
>
    <button
        onClick={[Function]}
    >
        Pat me!
    </button>
</div>,
  "nodes": Array [
    <div
      className="section"
>
      <button
            onClick={[Function]}
      >
            Pat me!
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Animal />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Animal {
        "_reactInternalInstance": [Circular],
        "clickEvent": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "clicked": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="section"
>
          <button
                    onClick={[Function]}
          >
                    Pat me!
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="section"
>
          <button
                    onClick={[Function]}
          >
                    Pat me!
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Animal />,
}
`;
